WordPress/Classes/ViewRelated/Stats/Extensions/Double+Stats.swift

```diff
@@ -13,6 +19,34 @@ extension Double {
         }
     }
 
+    private var decimalFormatter: NumberFormatter {
+         get {
+             let formatter = Double._decimalFormatter
+             // Show at least one digit after the decimal
+             formatter.minimumFractionDigits = 1
+             return formatter
+         }
+     }
+
+    private static var _units: [Unit] {
+        get {
+            var unitsArray: [Unit] = []
+            unitsArray.append(Unit(abbreviation: "K", name: "thousand"))
+            unitsArray.append(Unit(abbreviation: "M", name: "million"))
+            unitsArray.append(Unit(abbreviation: "B", name: "billion"))
+            unitsArray.append(Unit(abbreviation: "T", name: "trillion"))
+            unitsArray.append(Unit(abbreviation: "P", name: "quadrillion"))
+            unitsArray.append(Unit(abbreviation: "E", name: "quintillion"))
```

Comment:
Don't want to throw too much of a wrench in things, but aren't the large number groupings also different in many Locales? ([Indian](https://en.wikipedia.org/wiki/Indian_numbering_system), [Chinese](https://resources.allsetlearning.com/chinese/grammar/Big_numbers_in_Chinese)).

`NSNumberFormatter` has some handlers for some of this ([`secondaryGroupingSize`](https://developer.apple.com/documentation/foundation/numberformatter/1413348-secondarygroupingsize#)) but it's not really built in to the format strings at all.

I tried to play with augmenting `NumberFormatter` to do more of the work, which I think slightly simplifies the code but doesn't really solve this problem you're mentioning (note that the rounding change needs a bit of finagling for the higher numbers). Maybe setting the `positiveSuffix` / `positivePrefix` appropriately might be cleaner than swapping format string ordering? 🤷‍♂ 

```swift
extension Double {
        
    var abbreviated: String? {
        let numberFormatter = NumberFormatter()
        numberFormatter.locale = Locale(identifier: "ar_SA")
                
        // Convert to log1000
        guard self != 0 else { return "0" }
        let exp = Double(floor(log10(self) / log10(1000.0)))
        let mult = Double(pow(0.1, (exp * exp) + 2))
        
        let units: [String] = ["K", "M", "B", "T", "P", "E"]
        
        numberFormatter.multiplier = NSNumber(value: mult)//NSNumber(value: rounding/1000)
        numberFormatter.positiveFormat = "#,##0.#"
        numberFormatter.positiveSuffix = units[units.index(0, offsetBy: Int(exp) - 1)]
        
        return numberFormatter.string(for: self)
    }
}
```

Reply:
@bjtitus Thanks for jumping in! 

> but doesn't really solve this problem you're mentioning

Not sure I understand. Would you mind pointing me to which one isn't solved by your code? 😄 Sorry, I realized I wrote a very long message in https://github.com/wordpress-mobile/WordPress-iOS/pull/13214#r365284167. 

> but aren't the large number groupings also different in many Locales? (Indian, Chinese)

<img src="https://user-images.githubusercontent.com/198826/72173588-26a96200-3395-11ea-99d2-a96d208faf9c.png" width="420">

OMG. This got much more complicated. 😱 I really wish `NSNumberFormatter` handles all of these for us. 

I think we can consider this out of scope for now and stick with the `["K", "M", "B", "T", "P", "E"]`groupings. What do you think, @bjtitus? 

> Maybe setting the positiveSuffix / positivePrefix appropriately might be cleaner than swapping format string ordering?

This looks like a decent solution. However,  it doesn't seem to solve the possible spacing between the values. I think this is the _unsolved problem_ that you mentioned.

Also, I think this would require us to check if the device's language is RTL. Which can be another point of failure. I'd still probably prefer to leave the decision to the language translators.

Reply:
@shiki 

> Would you mind pointing me to which one isn't solved by your code?

Sure! I was referring to the fact that my code didn’t include the RTL switching you had mentioned:

> It turns out that the abbreviation should be on the left of the number for Arabic.

The prefix/suffix solution I mentioned was just a possible “easier” solution.

> it doesn't seem to solve the possible spacing between the values.

Yeah, I played around with this and thought the locale formatting might properly space things out but I can’t figure that one out.

> I think we can consider this out of scope for now and stick with the ["K", "M", "B", "T", "P", "E"]groupings.

I completely agree. I think we should probably be cautious going too far down this rabbit hole.

> I really wish NSNumberFormatter handles all of these for us.

🧚🏼‍♀️ They do this with Date and Measurement formatters. Seems like there could be a short/long formatting option added for simple counting units. I’ll file a feedback request.